sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -161,71 +160,21 @@ public String getDefaultLanguage() {
         try {
             Objects.requireNonNull(document, "'document' cannot be null.");
             return detectLanguageBatch(Collections.singletonList(document), countryHint, null)
-                .map(detectLanguageResult -> {
-                    if (detectLanguageResult.isError()) {
-                        throw logger.logExceptionAsError(toTextAnalyticsException(detectLanguageResult.getError()));
+                .map(detectLanguageResultCollection ->  {
+                    for (DetectLanguageResult detectLanguageResult : detectLanguageResultCollection) {
+                        if (detectLanguageResult.isError()) {
+                            throw logger.logExceptionAsError(toTextAnalyticsException(detectLanguageResult.getError()));
+                        }
+                        return detectLanguageResult.getPrimaryLanguage();
                     }
-                    return detectLanguageResult.getPrimaryLanguage();
-                }).last();
+                    // this step should never be executed, if executed, we get an empty collection above.
+                    return new DetectedLanguage(null, null, -1, null);
```

Comment:
Why do we have this in all the response mappers? Also, the comment is confusing. Either it gets executed or not. If it is expected to not be executed, then we should throw an exception when it does.

Reply:
Throw an exception will make java inconsistency with other languages.

Reply:
could you expand on what the scenario is here? not sure why we need to create this fake object. 
How will the user consume/use this?

Reply:
@maririos  It is what we discussed afternoon. .NET return it as it is. 

The scenario is when user gives a single input document, internally, it called the batch API.
And then we get the batch output result collection, which in here, `DetectLanguageResultCollection`, this class extends `IterableStream<DetectLanguageResult>`,  So the `DetectLanguageResultCollection` should has only one `DetectLanguageResult`. but the fake object created here should not be executed, since `DetectLanguageResultCollection` should has only one.

Reply:
@Shawn can be handled similarly https://github.com/Azure/azure-sdk-for-java/pull/11318/files#diff-8d0c97bfcab02df81ac3a72abd26a2a2R78-R89